File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int i=0, j=nums.size()-1;
        int n=nums.size();
        vector<int> res(n, 0);
        int pos=n-1;
        while(i<=j){
            if(nums[i]*nums[i]<nums[j]*nums[j]){
                res[pos--]=nums[j]*nums[j];
                j--;
            }
            else{
                res[pos--]=nums[i]*nums[i];
                i++;
            }
        }
        return res;
    }
};

File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        i, j=0, len(nums)-1
        n=len(nums)
        res=[0]*n
        pos=len(nums)-1

        while i<=j:
            if nums[i]*nums[i]<nums[j]*nums[j]:
                res[pos]=nums[j]*nums[j]
                j-=1
            else:
                res[pos]=nums[i]*nums[i]
                i+=1
            pos-=1
        return res